lib/previews.js

```diff
@@ -1,348 +1,379 @@
+      const sites_with_source_app = (await routes.list(pg_pool, [source_app_uuid])).filter((s) => s.site.preview === null);
+      const unique_site_names = sites_with_source_app.reduce((acc, val) => {
+        if (acc.indexOf(val.site.domain === -1)) {
+          return acc.concat([val.site.domain]);
+        }
+        return acc;
+      }, []);
+      await Promise.all(unique_site_names.map(async (site) => {
         // determine the new name for the site
-        let site_to_create = site.split('.').map((x, i) => {
-          if(i === 0) {
-            return x.substring(0, max_subdomain_length - suffix.length) + suffix
-          } else {
-            return x 
+        const site_to_create = site.split('.').map((x, i) => {
+          if (i === 0) {
+            return x.substring(0, max_subdomain_length - suffix.length) + suffix;
           }
-        }).join('.')
-        
-        let source_site_info = sites_with_source_app.filter((x) => x.site.domain === site)[0]
-        let new_site = await sites.create(pg_pool, source_site_info.site.compliance.indexOf('internal') !== -1, source_site_info.site.region, site_to_create)
-        new_sites.push(new_site)
-        await sites.enable_preview(pg_pool, new_site.id, preview_uuid)
-        let site_routes = await routes.list_by_site(pg_pool, [site]);
+          return x;
+        }).join('.');
+
+        const source_site_info = sites_with_source_app.filter((x) => x.site.domain === site)[0];
+        const new_site = await sites.create(pg_pool, source_site_info.site.compliance.indexOf('internal') !== -1, source_site_info.site.region, site_to_create);
+        new_sites.push(new_site);
+        await sites.enable_preview(pg_pool, new_site.id, preview_uuid);
+        const site_routes = await routes.list_by_site(pg_pool, [site]);
         await Promise.all(site_routes.map(async (route) => {
-          let target_app_id = source_app_uuid === route.app.id ? app_setup_status.app.id : route.app.id
+          const target_app_id = source_app_uuid === route.app.id ? app_setup_status.app.id : route.app.id;
           try {
-            return await routes.create(pg_pool, new_site.id, target_app_id, route.source_path, route.target_path, true)
+            return await routes.create(pg_pool, new_site.id, target_app_id, route.source_path, route.target_path, true);
           } catch (e) {
-            console.error(`Error: unable to create route for site ${new_site.id} and source app ${target_app_id} with path ${route.source_path} to ${route.target_path}`)
-            console.error(e.message)
-            console.error(e.stack)
+            console.error(`Error: unable to create route for site ${new_site.id} and source app ${target_app_id} with path ${route.source_path} to ${route.target_path}`);
+            console.error(e.message);
+            console.error(e.stack);
           }
-        }))
-        let non_pending_routes = site_routes.filter((x) => x.pending !== true)
+          return undefined;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unexpected use of undefined.](https://app.codacy.com/manual/Akkeris/controller-api/pullRequest?prid=5244943)